config/routes.rb

```diff
@@ -3,8 +3,14 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
-  resources :users do
-    resources :posts, only: [:index, :show], controller: 'posts'
-  end
   
-end
+  get '/users/:user_id/posts(.:format)', to: 'posts#index'
+  get  '/users/:user_id/posts(.:format)', to:  'posts#index'
+  get  '/users/:user_id/posts/:id(.:format)', to: 'posts#show'
+  get  '/users(.:format)', to: 'users#index'
+  get  '/users/:id(.:format)', to: 'users#show'
```

Comment:
- [ ] [OPTIONAL] Kindly update your previous implementation to be:
```ruby 
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```
and it will handle the 4 required URLs.  as this syntax is much easier to read and more professional.

Reply:
Thanks @gemmen29 for this.